### PR TITLE
[MAINTENANCE] Ensure that new usage statistics schema changes are backwards compatible

### PR DESCRIPTION
**Pull request by cdkini**
New persistent identifiers should not be mandatory to allow for old GX-version users to still provide valid payloads

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/integration/usage_statistics/test_usage_statistics_messages.py [hunk 1]
```diff
@@ -62,8 +62,6 @@ def test_generate_messages_with_defaults():
         "event_time": "2020-08-04T22:50:58.837Z",
         "data_context_id": TEST_DATA_CONTEXT_ID,
         "data_context_instance_id": TEST_DATA_CONTEXT_INSTANCE_ID,
-        "mac_address": TEST_MAC_ADDRESS,
-        "oss_id": TEST_OSS_ID,
```
Comment (cdkini) [inline tests/integration/usage_statistics/test_usage_statistics_messages.py:66]: We shouldn't update our old examples - I've deleted all these new identifier additions and then added a few test cases with the persistent values (making sure to utilize a modern version of GX, a recent event time, etc to model a real-life event)

File: great_expectations/core/usage_statistics/schemas.py [hunk 2]
```diff
@@ -1212,8 +1212,6 @@
         "event_time",
         "data_context_id",
         "data_context_instance_id",
-        "mac_address",
-        "oss_id",
```
Comment (cdkini) [inline great_expectations/core/usage_statistics/schemas.py:1216]: These are not mandatory - if a user is using something before v17, they won't have these

**Review by anthonyburdi — APPROVED** [review]
LGTM!

File: great_expectations/core/usage_statistics/schemas.py (hunk 2 above)
Comment (tannerbeam) [reply great_expectations/core/usage_statistics/schemas.py:1216]: i will need to modify the schema so these values can be captured. are we launching 0.17.0 this week?

File: tests/integration/usage_statistics/test_usage_statistics_messages.py [hunk 3]
```diff
@@ -418,9 +457,24 @@ def test_generate_messages_with_defaults():
             "event_time": "2020-06-25T16:08:16.030Z",
             "data_context_id": TEST_DATA_CONTEXT_ID,
             "data_context_instance_id": TEST_DATA_CONTEXT_INSTANCE_ID,
+            "ge_version": "0.11.9.manual_testing",
+            "x-forwarded-for": "00.000.00.000, 00.000.000.000",
```
Comment (tannerbeam) [inline tests/integration/usage_statistics/test_usage_statistics_messages.py:519]: this values are anonymized. you'll have to check the anonymizers to see what kind of values they are, but they're always anonymized and never a real IP address.

**Review by tannerbeam — APPROVED** [review]
pumped to try this out. nice work!

File: great_expectations/core/usage_statistics/schemas.py (hunk 2 above)
Comment (cdkini) [reply great_expectations/core/usage_statistics/schemas.py:1216]: yes it should be released Thursday - does that work?

**Pull request merged**